sdk/typescript/src/types/common.ts

```diff
@@ -32,22 +35,52 @@ export function isValidTransactionDigest(value: string): value is TransactionDig
 // which uses the Move account address length
 // https://github.com/move-language/move/blob/67ec40dc50c66c34fd73512fcc412f3b68d67235/language/move-core/types/src/account_address.rs#L23 .
 
-const SUI_ADDRESS_LENGTH = 20;
+export const SUI_ADDRESS_LENGTH = 20;
 export function isValidSuiAddress(value: string): value is SuiAddress {
-  return isHex(value) &&
-    getHexByteLength(value) === SUI_ADDRESS_LENGTH;
+  return isHex(value) && getHexByteLength(value) === SUI_ADDRESS_LENGTH;
 }
 
 export function isValidSuiObjectId(value: string): boolean {
   return isValidSuiAddress(value);
 }
 
+/**
+ * Perform the following operations:
+ * 1. Make the address lower case
+ * 2. Prepend `0x` if the string does not start with `0x`.
+ * 3. Add more zeros if the length of the address(excluding `0x`) is less than `SUI_ADDRESS_LENGTH`
+ *
+ * WARNING: if the address value itself starts with `0x`, e.g., `0x0x`, the default behavior
+ * is to treat the first `0x` not as part of the address. The default behavior can be overridden by
+ * setting `forceAdd0x` to true
+ *
+ */
+export function normalizeSuiAddress(
+  value: string,
+  forceAdd0x: boolean = false
+): SuiAddress {
+  let address = value.toLowerCase();
+  if (!forceAdd0x && address.startsWith('0x')) {
+    address = address.slice(2);
+  }
+  const numMissingZeros = (SUI_ADDRESS_LENGTH - getHexByteLength(address)) * 2;
+  if (numMissingZeros <= 0) {
+    return '0x' + address;
+  }
+  return '0x' + '0'.repeat(numMissingZeros) + address;
```

Comment:
Can probably simplify to `0x${address.padStart(SUI_ADDRESS_LENGTH, '0')}`, which removes the need for the `if` entirely.